File: cli/internal/modules/dockerfile/bun.go
package dockerfile

import (
	"fmt"

	"github.com/joacohbc/my-devcontainer-installer/cli/internal/core"
)

var BunModule = &DockerfileModuleSpec{
	ID:       "bun",
	Label:    "Bun",
	Category: core.CategoryRuntime,
	Render: func(opts map[string]any) string {
		return fmt.Sprintf(`##
## BUN
##
RUN su - devuser -c "curl -fsSL https://bun.sh/install | bash"
%s
`, emitShellInit(".bun_init.sh", []string{
			`export BUN_INSTALL="$HOME/.bun"`,
			`export PATH="$BUN_INSTALL/bin:$PATH"`,
		}))
	},
}

File: cli-ts-legacy/src/modules/dockerfile/bun.ts
import type { DockerfileModule } from '@/core/types.js';
import { emitShellInit } from '@/modules/dockerfile/shell-init.js';

export const bunModule: DockerfileModule = {
  id: 'bun',
  label: 'Bun',
  category: 'runtime',
  render() {
    return `##
## BUN
##
RUN su - devuser -c "curl -fsSL https://bun.sh/install | bash"
${emitShellInit('.bun_init.sh', [
  'export BUN_INSTALL="$HOME/.bun"',
  'export PATH="$BUN_INSTALL/bin:$PATH"',
])}
`;
  },
};
